src/main/java/org/skife/jdbi/v2/PreparedBatch.java

```diff
@@ -118,7 +134,12 @@ public PreparedBatch define(final Map<String, ? extends Object> values)
         return executeAndGenerateKeys(new SingleColumnMapper<GeneratedKeyType>(mapper));
     }
 
-    private <Result> Object internalBatchExecute(QueryResultMunger<Result> munger) {
+    public <GeneratedKeyType> GeneratedKeys<GeneratedKeyType> executeAndGenerateKeys(ResultColumnMapper<GeneratedKeyType> mapper,
```

Comment:
Another thought, should this use a `SingleColumnMapper#byLabel`?  Or maybe that doesn't matter...